pkg/authorization/authorizer/cache/authorizer_test.go

```diff
@@ -20,35 +18,34 @@ func TestAuthorizer(t *testing.T) {
 
 func TestCacheKey(t *testing.T) {
 	tests := map[string]struct {
-		Context kapi.Context
-		Attrs   authorizer.Action
+		Attrs authorizer.Attributes
 
 		ExpectedKey string
 		ExpectedErr bool
 	}{
 		"empty": {
-			Context:     kapi.NewContext(),
-			Attrs:       &authorizer.DefaultAuthorizationAttributes{},
-			ExpectedKey: `{"apiGroup":"","apiVersion":"","nonResourceURL":false,"resource":"","resourceName":"","subresource":"","url":"","verb":""}`,
+			Attrs:       authorizer.AttributesRecord{},
+			ExpectedKey: `{"apiGroup":"","apiVersion":"","name":"","namespace":"","path":"","readOnly":false,"resource":"","resourceRequest":false,"subresource":"","verb":""}`,
 		},
 		"full": {
-			Context: kapi.WithUser(kapi.WithNamespace(kapi.NewContext(), "myns"), &user.DefaultInfo{Name: "me", Groups: []string{"group1", "group2"}}),
-			Attrs: &authorizer.DefaultAuthorizationAttributes{
-				Verb:           "v",
-				APIVersion:     "av",
-				APIGroup:       "ag",
-				Resource:       "r",
-				Subresource:    "sub",
-				ResourceName:   "rn",
-				NonResourceURL: true,
-				URL:            "/abc",
+			Attrs: authorizer.AttributesRecord{
+				User:            &user.DefaultInfo{Name: "me", Groups: []string{"group1", "group2"}},
+				Verb:            "v",
+				Namespace:       "myns",
+				APIVersion:      "av",
+				APIGroup:        "ag",
+				Resource:        "r",
+				Subresource:     "sub",
+				Name:            "rn",
+				ResourceRequest: true,
```

Comment:
> doesn't really matter, but NonResourceURL: true is the opposite of ResourceRequest: true

Didn't matter in the test.  I thought it better to test something other than zero values.